backend/corpora/lambdas/api/v1/collection.py

```diff
@@ -68,3 +68,17 @@ def create_collection(body: object, user: str):
 
 def get_collection_dataset(dataset_uuid: str):
     raise NotImplementedError
+
+
+def delete_collection(collection_uuid: str, visibility: str, user: str):
+    if visibility == CollectionVisibility.PUBLIC.name:
+        return make_response(jsonify("Can not delete a public dataset"), 405)
+    db_session = g.db_session
+    collection = Collection.get_collection(db_session, collection_uuid, visibility, include_tombstones=True)
+    if not collection:
+        raise ForbiddenHTTPException()
```

Comment:
Is 403 the most accurate exception we can throw here? I feel like it's a little misleading especially since the default message is saying the user is not authorized.  Maybe a 404 is more accurate?